cmd/system/aws_cli.go

```diff
@@ -0,0 +1,317 @@
+package system
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/Masterminds/semver"
+	"github.com/alexellis/arkade/pkg/archive"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/get"
+	execute "github.com/alexellis/go-execute/v2"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallAWSCLI() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "aws-cli",
+		Short: "Install AWS CLI",
+		Long:  `Install AWS CLI for interacting with Amazon Web Services APIs.`,
+		Example: `  arkade system install aws-cli
+  arkade system install aws-cli --version <version>`,
+		SilenceUsage: true,
+	}
+
+	command.Flags().StringP("version", "v", "", "The version or leave blank to determine the latest available version")
+	command.Flags().String("path", "/usr/local/bin", "Installation path, where a aws cli subfolder will be created")
+	command.Flags().Bool("progress", true, "Show download progress")
+	command.Flags().StringP("work-dir", "w", "", "Working directory that installer files should be copied to (current directory if not supplied)")
+	command.Flags().Bool("run-installer", false, "Whether or not arkade should run the downloaded installer")
+
+	command.PreRunE = func(cmd *cobra.Command, args []string) error {
+		_, err := cmd.Flags().GetString("path")
+		if err != nil {
+			return err
+		}
+
+		_, err = cmd.Flags().GetBool("progress")
+		if err != nil {
+			return err
+		}
+
+		_, err = cmd.Flags().GetString("work-dir")
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		installPath, _ := cmd.Flags().GetString("path")
+		version, _ := cmd.Flags().GetString("version")
+		progress, _ := cmd.Flags().GetBool("progress")
+		workDir, _ := cmd.Flags().GetString("work-dir")
+		runInstaller, _ := cmd.Flags().GetBool("run-installer")
+		fmt.Printf("Installing AWS CLI to %s\n", installPath)
+
+		installPath = strings.ReplaceAll(installPath, "$HOME", os.Getenv("HOME"))
+
+		if err := os.MkdirAll(installPath, 0755); err != nil && !os.IsExist(err) {
+			fmt.Printf("Error creating directory %s, error: %s\n", installPath, err.Error())
+		}
+
+		arch, osVer := env.GetClientArch()
+
+		if cmd.Flags().Changed("arch") {
+			arch, _ = cmd.Flags().GetString("arch")
+		}
+
+		if version == "" {
+			v, err := getAWSCLIVersion("aws", "aws-cli")
```

Comment:
Can't we replace this function + call to `FindGitHubRelease` from the `get` package?

That'll also cut the `go-git` from the dependency list.

Reply:
I believe there was a reason for this, and it is that the AWS CLI doesn't publish GitHub Releases, but only supplies tags that refer to a given release. I'd be happy to replace this with something else to get rid of that though? Maybe just raw HTTP Get to retrieve the tags?

Reply:
I've removed the dependency on `go-git`, but still the tags are the thing we'd need here I think?

Reply:
Hmm...you are right, the tags are waaay up-to-date then the releases page, releases should be the way to go.

Also I really like the API implementation!